tests/Fixer/ArrayNotation/ArraySingleMultiLineFixerTest.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ArrayNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ArrayNotation\ArraySingleMultiLineFixer
+ */
+final class ArraySingleMultiLineFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input, ?array $config): void
+    {
+        if (null !== $config) {
+            $this->fixer->configure($config);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases(): iterable
+    {
+        yield 'array square brace open' => [
+            '<?php $a = [
+1,
+2,
+3
+];',
+            '<?php $a = [1,2,3];',
+            ['element_count' => 3, 'inner_length' => 10000, 'conditions' => 'any'],
+        ];
+
```

Comment:
shouldn't we fix following case too?
```
        yield 'array square brace open2' => [
            '<?php $a = [
1,
2,
3
];',
            '<?php $a = [1,2,
3];',
            ['element_count' => 3, 'inner_length' => 10000, 'conditions' => 'any'],
        ];
```

Reply:
The fixer only touches array's to are written on a single line. 
Touch any array is very easy and works, how ever I found many cases in many projects that look something like:

```php
<?php
class Foo {
    private static array \$tags = [
        'api', 'author', 'category', 'copyright', 'deprecated', 'example',
        'global', 'internal', 'license', 'link', 'method', 'package', 'param',
        'property', 'property-read', 'property-write', 'return', 'see',
        'since', 'subpackage', 'throws', 'todo', 'uses', 'var', 'version',
    ];
}
```
which people don't want single or completely multilined